sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -949,16 +949,22 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
-    val tableMetadata = catalog.getTableMetadata(table)
-
-    // TODO: unify this after we unify the CREATE TABLE syntax for hive serde and data source table.
-    val stmt = if (DDLUtils.isDatasourceTable(tableMetadata)) {
-      showCreateDataSourceTable(tableMetadata)
+    if (catalog.isTemporaryTable(table)) {
+      throw new AnalysisException(
+        s"SHOW CREATE TABLE is not supported on a temporary view: ${table.identifier}")
     } else {
-      showCreateHiveTable(tableMetadata)
-    }
+      val tableMetadata = catalog.getTableMetadata(table)
 
-    Seq(Row(stmt))
+      // TODO: unify this after we unify the
```

Comment:
Hi, @wangyum .
I know that this `TODO` is not generated in this PR, but this is a good change to make this to `IDed TODO`. It will help the other contributors pick up this issue. Could you file a JIRA for this and use that JIRA ID here, please?

Reply:
Done: https://issues.apache.org/jira/browse/SPARK-28692